### PR TITLE
feat: 🎸 check if compliance requirements are paused

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 

https://polymesh.atlassian.net/browse/DA-64

### Changelog / Description 

adds endpoint for checking if compliacne requirements are paused for an Asset

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/compliance/compliance-requirements.controller.ts [hunk 1]
```diff
@@ -266,4 +266,30 @@ export class ComplianceRequirementsController {
     const result = await this.complianceRequirementsService.modify(ticker, id, params);
     return handleServiceResult(result);
   }
+
+  @ApiOperation({
+    summary: 'Check if the requirements are paused',
+    description: 'This endpoint checks if the requirements are paused',
```
Comment (prashantasdeveloper) [inline src/compliance/compliance-requirements.controller.ts:272]:
```suggestion
    description: 'This endpoint checks if the compliance requirements are paused for a given ticker',
```

File: src/compliance/compliance-requirements.controller.ts [hunk 2]
```diff
@@ -266,4 +266,30 @@ export class ComplianceRequirementsController {
...
+    },
+  })
+  @ApiNotFoundResponse({
+    description: 'The Asset does not exist',
+  })
+  @Post('are-paused')
+  public async areRequirementsPaused(@Param() { ticker }: TickerParamsDto): Promise<boolean> {
+    const result = await this.complianceRequirementsService.arePaused(ticker);
+
+    return result;
```
Comment (prashantasdeveloper) [inline src/compliance/compliance-requirements.controller.ts:293]:
```suggestion
    return this.complianceRequirementsService.arePaused(ticker);
```

File: src/compliance/compliance-requirements.service.spec.ts [hunk 3]
```diff
@@ -237,4 +237,17 @@ describe('ComplianceRequirementsService', () => {
       });
     });
   });
+
+  describe('arePaused', () => {
+    it('should return ', async () => {
```
Comment (prashantasdeveloper) [inline src/compliance/compliance-requirements.service.spec.ts:242]:
```suggestion
    it('should return the Asset compliance requirement state ', async () => {
```

File: src/compliance/compliance-requirements.controller.ts [hunk 4]
```diff
@@ -266,4 +266,30 @@ export class ComplianceRequirementsController {
...
+  @ApiOkResponse({
+    description: 'Compliance Requirement status',
+    schema: {
+      type: 'boolean',
+    },
+  })
+  @ApiNotFoundResponse({
+    description: 'The Asset does not exist',
+  })
+  @Post('are-paused')
```
Comment (prashantasdeveloper) [inline src/compliance/compliance-requirements.controller.ts:289]:
```suggestion
  @Get('are-paused')
```

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

File: src/compliance/compliance-requirements.controller.ts [hunk 5]
```diff
@@ -266,4 +266,29 @@ export class ComplianceRequirementsController {
...
+  @ApiOkResponse({
+    description: 'Compliance Requirement status',
+    schema: {
+      type: 'boolean',
+    },
+  })
+  @ApiNotFoundResponse({
+    description: 'The Asset does not exist',
+  })
+  @Get('are-paused')
```
Comment (polymath-eric) [inline src/compliance/compliance-requirements.controller.ts:290]: I'd say something like:
```suggestion
  @Get('/status')
```
and make the return an object with a key.

Generally we want to routes to be REST like, with the exception being for actions.

File: src/compliance/compliance-requirements.controller.ts [hunk 6]
```diff
@@ -266,4 +266,29 @@ export class ComplianceRequirementsController {
...
+  public async areRequirementsPaused(@Param() { ticker }: TickerParamsDto): Promise<boolean> {
```
Comment (polymath-eric) [inline src/compliance/compliance-requirements.controller.ts:291]:
```suggestion
  public async areRequirementsPaused(@Param() { ticker }: TickerParamsDto): Promise<ComplianceStatusModel> {
```
and declare it as a model.

Keep the scope to only being pause or not for now, but allow it to be extended and a bit more generic.

File: src/compliance/mocks/compliance-requirements.mock.ts [hunk 7]
```diff
@@ -24,3 +31,26 @@ export class MockComplianceRequirements {
...
+    {
+      id: new BigNumber(1),
+      conditions: [
+        {
+          type: ConditionType.IsPresent,
+          claim: {
+            type: ClaimType.Accredited,
+            scope: {
+              type: ScopeType.Identity,
+              value: 'Ox6'.padEnd(66, '0'),
```
Comment (polymath-eric) [inline src/compliance/mocks/compliance-requirements.mock.ts:46]: (note) if my consts PR gets merged, update these.

If this gets merged in first I will update in that PR.

**Review by polymath-eric — CHANGES_REQUESTED** [review]
Code looks good.

I think the route should be a noun since it is a GET and we should return a model to help with documentation and make it extensible.

**Review by polymath-eric — APPROVED** [review]

File: src/compliance/models/compliance-status.model.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import { FromBigNumber } from '~/common/decorators/transformation';
+
+export class ComplianceStatusModel {
+  @ApiProperty({
+    description: 'Are the compliance requirements paused',
+    type: 'boolean',
+    example: true,
+  })
+  @FromBigNumber()
+  readonly arePaused: boolean;
```
Comment (prashantasdeveloper) [inline src/compliance/models/compliance-status.model.ts:14]:
```suggestion
  @ApiProperty({
    description: 'Indicator to know if compliance requirements are paused or not',
    type: 'boolean',
    example: true,
  })
  readonly paused: boolean;
```

Comment (prashantasdeveloper) [reply src/compliance/models/compliance-status.model.ts:14]: `@FromBigNumber` is only needed when return type is BigNumber. Feels like it got copied over

File: src/compliance/compliance-requirements.service.spec.ts (hunk 3 above)
Comment (prashantasdeveloper) [reply src/compliance/compliance-requirements.service.spec.ts:244]: This was not updated

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**